src/coreclr/jit/importercalls.cpp

```diff
@@ -4165,21 +4165,29 @@ GenTree* Compiler::impIntrinsic(CORINFO_CLASS_HANDLE    clsHnd,
             case NI_System_Threading_Interlocked_Or:
             case NI_System_Threading_Interlocked_And:
             {
+                var_types actualType = genActualType(callType);
+#if defined(TARGET_X86)
+                // On x86, TYP_LONG is not supported as an intrinsic
+                if (actualType == TYP_LONG)
+                {
+                    break;
+                }
+#endif
+                // Only expand intrinsic for TYP_INT and TYP_LONG; fallback to managed for other types
```

Comment:
@egorbo Is this a permanent limitation or a TODO? I would expect the smaller sizes to work as intrinsics on x86/x64 at least.

Reply:
@jkotas I wanted to look at it separately myself, since currently `Interlocked.Or` and `.And` only exist for int/uint/long/ulong. So I want to check on arm64 and, perhaps, LA64/RISC-V if they accelerate it.